tests/load/locustfiles/locustfile.py

```diff
@@ -50,211 +72,282 @@ def _(environment, **kwargs):
+    WEBSOCKET_HEADERS: dict[str, str] = {
+        "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv:61.0) "
+        "Gecko/20100101 Firefox/61.0"
+    }
+
     def __init__(self, environment) -> None:
         super().__init__(environment)
-        self.uaid: str = ""
         self.channels: dict[str, str] = {}
-        self.ws: Any = None
-        self.headers = {"TTL": "60", "Content-Encoding": "aes128gcm", "Topic": "aaaa"}
-        self.encrypted_data = base64.urlsafe_b64decode(
-            "TestData"
-            + "".join(
-                [
-                    random.choice(string.ascii_letters + string.digits)
-                    for i in range(0, random.randrange(1024, 4096, 2) - 8)
-                ]
-            )
-            + "=="
-        )
+        self.hello_record: HelloRecord | None = None
+        self.notification_records: list[NotificationRecord] = []
+        self.register_records: list[RegisterRecord] = []
+        self.uaid: str = ""
+        self.ws: WebSocketApp | None = None
+        self.ws_greenlet: Greenlet | None = None
 
     def wait_time(self):
         return self.environment.autopush_wait_time(self)
 
     def on_start(self) -> Any:
-        self.connect()
-        if not self.ws:
-            self.stop()
-
-        self.hello()
-        if not self.uaid:
-            self.stop()
-
-        self.register()
-        if not self.channels:
-            self.stop()
+        """Called when a User starts running."""
+        self.ws_greenlet = gevent.spawn(self.connect)
 
     def on_stop(self) -> Any:
+        """Called when a User stops running."""
         if self.ws:
-            self.disconnect()
-
-    def _time_event(self, name: str) -> TimeEvent:
-        return TimeEvent(self, name)
+            self.ws.close()
+        if self.ws_greenlet:
+            gevent.kill(self.ws_greenlet)
 
-    def connect(self) -> None:
-        self.ws = create_connection(
-            self.environment.parsed_options.websocket_url,
-            header={"Origin": "http://localhost:1337"},
-            timeout=5,  # timeout defaults to None
-        )
+    def on_ws_open(self, ws: WebSocketApp) -> None:
+        """Called when opening a WebSocket.
 
-    def disconnect(self) -> None:
-        if self.ws:
-            self.ws.close()
-        self.ws = None
+        Args:
+            ws: WebSocket class object
+        """
+        self.send_hello(ws)
 
-    def hello(self) -> None:
-        """Send a 'hello' message to Autopush.
+    def on_ws_message(self, ws: WebSocketApp, data: str) -> None:
+        """Called when received data from a WebSocket.
 
-        Connections must say hello after connecting to the server, otherwise the connection is
-        quickly dropped.
+        Args:
+            ws: WebSocket class object
+            data: utf-8 data received from the server
+        """
+        message: Message = self.recv(data)
+        if isinstance(message, HelloMessage):
```

Comment:
FYI (we don't need to change this) Python's match statement can replace multiple `isinstance` checks like this. It looks like
```suggestion
        match message:
            case HelloMessage():
```

Reply:
mypy is raising the following errors when I make the switch:
<img width="762" alt="image" src="https://github.com/mozilla-services/autopush-rs/assets/3422688/08205107-2952-4a56-ad9d-ca99be49172a">
I'm inclined to find the solution another day, unless you feel strongly about the match-case?